### PR TITLE
Add C# 9 test cases for S1226,  S1206, S3447, S3466, S4050, S4069, S1123, S3937 and S3459

### PR DESCRIPTION
**Pull request by costin-zaharia-sonarsource**
Related to #3668

### CODE REVIEW

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/ParameterAssignedTo.CSharp9.cs [hunk 1]
```diff
@@ -0,0 +1,21 @@
+﻿using System;
+
+static void Foo(int x)
+{
+    Action<int, int> discard = (_, _) => _ = 10; // Compliant
```
Comment (andrei-epure-sonarsource) [inline sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/ParameterAssignedTo.CSharp9.cs:5]: also a test would be nice
```
    Action<int> underscoreName = (_) => _ = 10; // Noncompliant
```

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/ParameterAssignedTo.CSharp9.cs [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+    Action<int, int> foo = (a, b) => a = 10; // Noncompliant
+}
+
+public record Record
```
Comment (andrei-epure-sonarsource) [inline sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/ParameterAssignedTo.CSharp9.cs:10]: how does this rule behave with `set` and `init` for Properties, if `value` is assigned?
IMO it should raise

**Review by andrei-epure-sonarsource — APPROVED** [review]
LGTM, I suggest to add two more tests for ParameterAssignedTo before merging

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/ParameterAssignedTo.CSharp9.cs (hunk 2 above)
Comment (costin-zaharia-sonarsource) [reply sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/ParameterAssignedTo.CSharp9.cs:10]: Nice! I've added the test cases. 

It works as expected but I did not think at all that you might assign to `value`.

**Pull request merged**